### PR TITLE
remove cgi dependency

### PR DESCRIPTION
**Pull request by stathis-alexander**
Ruby 4 no longer includes CGI as a standard library, you can read more about the motivation for that change [here](https://bugs.ruby-lang.org/issues/21258). This change removes CGI from this gem, allowing Ruby 4 applications to avoid the extra CGI dependency completely, by replacing the one usage of `CGI.parse` with `URI.decode_www_form`. 

This change is potentially breaking for consumers, as it changes the semantics of `;` in query strings. Previously, `;` was treated as a parsing break, allowing query strings of the form:
```
https://url?a=1;b=2
```
to be parsed as separate `a` and `b` key-value pairs. Now, `;` is ignored and only `&` breaks.

### CODE REVIEW

File: lib/anthropic/internal/util.rb [hunk 1]
```diff
@@ -318,7 +318,9 @@ class << self
         #
         # @return [Hash{String=>Array<String>}]
         def decode_query(query)
-          CGI.parse(query.to_s)
+          URI.decode_www_form(query.to_s).each_with_object({}) do |(key, value), out|
```
Comment (stathis-alexander) [inline lib/anthropic/internal/util.rb:321]: This could potentially be a breaking change, as it changes the treatment of `;`. Previously, the parser handled additional `;` like `&`. Now it does not. Internally, `;` is not used for this purpose.

**Comment by stathis-alexander** [comment]
Hi @dtmeadows-ant , mind taking a look at this PR when you get a chance? No rush.

**Comment by dtmeadows-ant** [comment]
Hey @stathis-alexander thanks so much for the PR. Will take a deeper look shortly

File: lib/anthropic/internal/util.rb (hunk 1 above)
Comment (dtmeadows-ant) [reply lib/anthropic/internal/util.rb:321]: Definitely good to callout the change. Overall, the SDK controls basically all of the real use cases of this so I'm not super concerned about the edge cases here changing. Only `base_url` and raw `path:` strings reach this, and our own paths are all `?beta=true`. Let's just make sure to call out the behavior changes in the description / commit.

File: test/anthropic/internal/util_test.rb [hunk 2]
```diff
@@ -175,6 +175,21 @@ def test_non_time_scalars_unchanged
 end
 
 class Anthropic::Test::UtilUriHandlingTest < Minitest::Test
+  def test_decode_query
```
Comment (dtmeadows-ant) [inline test/anthropic/internal/util_test.rb:178]: Could we add the cases whose behavior changed vs `CGI.parse` so the chosen semantics are pinned: `"a"`, `"a&b=1"`, `"a=1;b=2"`, `"a=1&&b=2"`, `"&a=1"`, and a non-UTF-8 escape like `"q=%E9"`?

File: test/anthropic/internal/util_test.rb [hunk 3]
```diff
@@ -293,25 +308,55 @@ def test_jsonl_content
 end
 
 class Anthropic::Test::UtilFormDataEncodingTest < Minitest::Test
-  class FakeCGI < CGI
+  # Minimal multipart reader so these tests do not depend on the `cgi` gem.
```
Comment (dtmeadows-ant) [inline test/anthropic/internal/util_test.rb:311]: I think it might be better to keep `cgi` here but just as a dev/test dependency?

**Review by dtmeadows-ant — CHANGES_REQUESTED** [review]
Overall looking good! Mind just updating some of the tests areas I commented on?

**Comment by stathis-alexander** [comment]
Thanks for looking! Will get to this today or tomorrow.

File: Gemfile [hunk 4]
```diff
@@ -29,6 +29,7 @@ end
 
 group :development, :test do
   gem "async"
+  gem "cgi"
```
Comment (stathis-alexander) [inline Gemfile:32]: We weren't using the test matrix for CGI, just `base64` and `connection_pool`.

File: lib/anthropic/internal/util.rb (hunk 1 above)
Comment (stathis-alexander) [reply lib/anthropic/internal/util.rb:321]: I updated the PR description. I assume that will be used when this is merged?

**Comment by stathis-alexander** [comment]
@dtmeadows-ant Thank you for the review. I believe I have addressed your comments. Let me know if any further changes are required.

**Review by dtmeadows-ant — CHANGES_REQUESTED** [review]
Thanks for the quick turnaround. Two things before this can go in:
1. CI is red on your fork for this commit: test_file_encode / test_hash_encode fail with undefined method '[]' for FakeCGI. Adding require "cgi" at the top of test/anthropic/internal/util_test.rb should fix it.
2. Gemfile.lock moved cgi 0.5.1 → 0.3.6. Could you bundle lock --update cgi so it stays on 0.5.x?

**Comment by stathis-alexander** [comment]
Sorry about that, should be good to go now - I don't have the mock server setup and so `bundle exec rake test` doesn't run clean. 

Also - I ran `bundle exec rake format` and found a bunch of unrelated changes.

**Review by dtmeadows-ant — APPROVED** [review]
